api/v1/helpers.go

```diff
@@ -572,3 +572,15 @@ func (v *VerticaDB) GetEncryptSpreadComm() string {
 func (v *VerticaDB) IsKSafetyCheckStrict() bool {
 	return vmeta.IsKSafetyCheckStrict(v.Annotations)
 }
+
+// IsValidRestorePointPolicy will true if the RestorePointPolicy is properly specified,
+// i.e. having a non-empty archive, and either a valid index or a valid id (but not both)
+func (r *RestorePointPolicy) IsValidRestorePointPolicy() bool {
+	return r != nil && r.Archive != "" && ((r.Index > 0) != (r.ID != ""))
+}
+
+// IsRestoreConfig will return whether the vdb is configured to initialize by reviving from
+// a restore point in an archive
+func (v *VerticaDB) IsRestoreConfig() bool {
+	return v.Spec.InitPolicy == CommunalInitPolicyRevive && v.Spec.RestorePoint != nil && v.Spec.RestorePoint.Archive != ""
```

Comment:
```suggestion
	return v.Spec.InitPolicy == CommunalInitPolicyRevive && v.Spec.RestorePoint != nil
```
We are checking archive later so do we need this here. With this the webhook will never let the user know that their CR is wrong because Archive is empty.